src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -4339,23 +4339,18 @@ private BoundExpression BindValueConditionalOperator(ConditionalExpressionSyntax
                 return new BoundUnconvertedConditionalOperator(node, condition, trueExpr, falseExpr, constantValue, noCommonTypeError, hasErrors: constantValue?.IsBad == true);
             }
 
-            TypeSymbol type;
             bool hasErrors;
             if (bestType.IsErrorType())
             {
                 trueExpr = BindToNaturalType(trueExpr, diagnostics, reportNoTargetType: false);
                 falseExpr = BindToNaturalType(falseExpr, diagnostics, reportNoTargetType: false);
-                type = bestType;
                 hasErrors = true;
             }
             else
             {
                 trueExpr = GenerateConversionForAssignment(bestType, trueExpr, diagnostics);
                 falseExpr = GenerateConversionForAssignment(bestType, falseExpr, diagnostics);
                 hasErrors = trueExpr.HasAnyErrors || falseExpr.HasAnyErrors;
-                // If one of the conversions went wrong (e.g. return type of method group being converted
-                // didn't match), then we don't want to use bestType because it's not accurate.
-                type = hasErrors ? CreateErrorType() : bestType;
```

Comment:
To fix the bug, it seems sufficient to either take this change (to use the best common type in binding always), or the change in NullableWalker (to use best common in nullable analysis always). But both changes seemed appropriate to me.

Reply:
Let me check my understanding.

1. In this path, bestType is not null and not ErrorType. If one of the conditional expr operands had an ErrorType, we would have inferred ErrorType as the bestType.
2. The removed lines here are checking if there is some error within the operands, which didn't cause them to have an ErrorType, per (1). We previously thought in this case that we need to change the conditional-expr type to ErrorType in this case. Now we are simply keeping the bestType which we already determined in the face of such errors.

Basically I'm not seeing what purpose the original check was serving, it feels like this change is just propagating along reasonable type information, which allows our assumptions about the bound tree shape to be met when we reach nullable analysis.

Reply:
I do see that there is basically a conversion check from the operands to the best type, which can fail, and I also find that a bit confusing. Perhaps my issue is that we are not checking specifically if the conversion had errors, we are checking for any kind of errors within the whole operand.

Essentially if one of the operands doesn't convert to the "best type", why was that chosen as the "best type"? Are there some invariants in best type analysis we can rely on in this regard?

Reply:
`BestTypeInferrer.InferBestTypeForConditionalOperator()` and `GenerateConversionForAssignment()` are both using `ClassifyImplicitConversionFromExpression()`, so it's not clear when the `GenerateConversionForAssignment()` calls above would fail if the best common type is not an error type.